core/src/org/pentaho/di/core/xml/XMLHandler.java

```diff
@@ -768,20 +768,19 @@ public static String getString() {
    */
   public static String addTagValue( String tag, String val, boolean cr, String... attributes ) {
     StringBuilder value = new StringBuilder( "<" );
-    Encoder encoder = ESAPI.encoder();
-    value.append( tag );
+    value.append( Encode.forXml( tag ) );
 
     for ( int i = 0; i < attributes.length; i += 2 ) {
-      value.append( " " ).append( encoder.encodeForXMLAttribute( attributes[i] ) ).append( "=\"" ).append(
-        attributes[i + 1] ).append( "\" " );
+      value.append( " " ).append( Encode.forXml( attributes[i] ) ).append( "=\"" ).append(
```

Comment:
I believe you need forXmlAttribute here. The rules for an XML element are very different than the encoder rules for an XML attribute. You need to accommodate this difference in a Unit Test if it's not already covered. https://www.w3.org/TR/xml/#NT-AttValue - note you cannot have ampersand or open angle brackets, and you can't have the same character used to open the attribute name ( either ' or " depending on how the attribute is formed )